作業目標:<br>
1. 敘述統計量計算<br>
2. 運用自定義函數apply

作業重點:<br>
1. 了解敘述統計量並解釋<br>
2. 運用apply時須注意自定義函數寫法

### 題目: 請對以下成績資料做分析

In [7]:
import pandas as pd
score_df = pd.DataFrame([[1,56,66,70], 
              [2,90,45,34],
              [3,45,32,55],
              [4,70,77,89],
              [5,56,80,70],
              [6,60,54,55],
              [7,45,70,79],
              [8,34,77,76],
              [9,25,87,60],
              [10,88,40,43]],columns=['student_id','math_score','english_score','chinese_score'])
score_df = score_df.set_index('student_id')
score_df

math_score  english_score  chinese_score
student_id                                          
1                   56             66             70
2                   90             45             34
3                   45             32             55
4                   70             77             89
5                   56             80             70
6                   60             54             55
7                   45             70             79
8                   34             77             76
9                   25             87             60
10                  88             40             43

### 1. 6號學生(student_id=6)3科平均分數為何?

In [14]:
score_df.mean(axis = 1).round(2)

student_id
1     64.00
2     56.33
3     44.00
4     78.67
5     68.67
6     56.33
7     64.67
8     62.33
9     57.33
10    57.00
dtype: float64

Ans: 6號學生3科平均分數為 56.33。

### 2. 6號學生3科平均分數是否有贏過班上一半的同學

In [19]:
score_df.mean(axis = 1).round(2).median()

59.83

Ans: 班上學生3科平均的中位數是 59.83 分 而6號學生3科平均為 56.33 分，因此並沒有贏過班上一半的同學。

### 3. 由於班上同學成績不好，所以學校統一加分，加分方式為開根號乘以十，請問6號同學3科成績分別是?


In [22]:
score_df.apply(lambda x : x**0.5*10).round(2)

math_score  english_score  chinese_score
student_id                                          
1                74.83          81.24          83.67
2                94.87          67.08          58.31
3                67.08          56.57          74.16
4                83.67          87.75          94.34
5                74.83          89.44          83.67
6                77.46          73.48          74.16
7                67.08          83.67          88.88
8                58.31          87.75          87.18
9                50.00          93.27          77.46
10               93.81          63.25          65.57

Ans: 數學: 77.46;英文: 73.48;國文: 74.16

### 4. 承上題，加分後各科班平均變多少

In [24]:
score_df.apply(lambda x : x**0.5*10).mean().round(2)

math_score       74.19
english_score    78.35
chinese_score    78.74
dtype: float64

Ans: 數學: 74.19;英文: 78.35;國文: 78.74